src/codegen/mod.rs

```diff
@@ -1302,11 +1420,50 @@ impl CodeGenerator for CompInfo {
                                 ::$prefix::mem::align_of::<$ident>());
                 let size = layout.size;
                 let align = layout.align;
+
+                let check_struct_align = if align > mem::size_of::<* mut ()>() {
+                    // FIXME when [RFC 1358](https://github.com/rust-lang/rust/issues/33626) ready
+                    None
+                } else {
+                    quote_item!(ctx.ext_cx(),
+                        assert_eq!($align_of_expr, $align);
+                    )
+                };
+
+                let check_field_offset = if item.is_opaque(ctx) ||
```

Comment:
Please add a `FIXME` comment pointing to #465 so people can understand the `base_members` check. It's probably worth moving this condition to its own variable. Something like:

```rust
let should_skip_field_offset_checks = item.is_opaque(ctx) || ...;
```